app/(main)/settings/team/actions.ts

```diff
@@ -480,3 +480,61 @@ export async function getAgentActivities({
 		};
 	}
 }
+
+export async function deleteTeam() {
+	try {
+		// Get current user's info and team
+		const currentTeam = await fetchCurrentTeam();
+		const currentUserRoleResult = await getCurrentUserRole();
+
+		// Check if user is admin
+		if (
+			!currentUserRoleResult.success ||
+			currentUserRoleResult.data !== "admin"
+		) {
+			throw new Error("Only admin users can delete teams");
+		}
+
+		// Check if team is on Free plan
+		if (isProPlan(currentTeam)) {
+			throw new Error("Only free plan teams can be deleted");
+		}
+
+		// Get current user's other teams count
+		const supabaseUser = await getUser();
+		const currentUserTeamsCount = await db
+			.select({
+				count: count(),
+			})
+			.from(teamMemberships)
+			.innerJoin(
+				supabaseUserMappings,
+				eq(teamMemberships.userDbId, supabaseUserMappings.userDbId),
+			)
+			.where(
+				and(
+					eq(supabaseUserMappings.supabaseUserId, supabaseUser.id),
+					ne(teamMemberships.teamDbId, currentTeam.dbId), // Exclude current team
+				),
+			);
+
+		// Check if user has other teams
+		if (currentUserTeamsCount[0].count === 0) {
+			throw new Error("Cannot delete your only team");
+		}
+
+		// Delete team and all related data with cascading
+		await db.delete(teams).where(eq(teams.dbId, currentTeam.dbId));
+
```

Comment:
Could you call `updateGiselleSession()` here?
I want to update teamDbId in cookie instead of deleted team's ID.

Reply:
@satococoa updated ♻️ [425ba4f](https://github.com/giselles-ai/giselle/pull/244/commits/425ba4f38e850acb7feb0a19c6113d1af91e7ab1)